src/cpu/flexcpu/simple_dataflow_cpu.cc

```diff
@@ -247,31 +346,24 @@ SimpleDataflowCPU::init()
     // TODO add post-construction initialization code
 }
 
-bool
+void
 SimpleDataflowCPU::requestDataAddrTranslation(const RequestPtr& req,
     ThreadContext* tc, bool write,
     TranslationCallback callback_func)
 {
     DPRINTF(SDCPUCoreEvent, "requestDataAddrTranslation()\n");
-    // enqueue if we're out of slots for simultaneous translations
-    // instTranslationReqs.push_back({req, callback_func});
-
-    // Retrieve itb from the ThreadContext, since TLBs are specific to threads
-    BaseTLB* dtb = tc->getDTBPtr();
-
-    // The event callback for when the translation completed comes through
-    // a BaseTLB::Translation object, so we pass that callback straight to the
-    // requestor's callback function by subclassing it.
-    BaseTLB::Translation* handler = new CallbackTransHandler(callback_func,
-                                                             true);
-
-    dtb->translateTiming(req, tc, handler,
-                         write ? BaseTLB::Write : BaseTLB::Read);
+    // delay if we're out of slots for simultaneous dtb translations
 
-    return true;
+    if (clockedDtbTranslation) {
```

Comment:
Why not always add an event and in one case call it with clockEdge(latency) and the other with curTick()?

I'm still not a fan of the callAtClockEdge function. I think it would make more sense for all of the functions to have their own events and schedule in each function.

Reply:
I think this was a thing we asked at one point, if this needed to be in its own event, or if it could be allowed to sit in the same call stack. I think we decided to have it in the same call stack for the time being because it was easier and probably faster, until a change was necessary...